bundles/org.eclipse.emfcloud.modelserver.common/src/org/eclipse/emfcloud/modelserver/common/patch/AbstractJsonPatchHelper.java

```diff
@@ -0,0 +1,716 @@
+      // Since we expect path as <id>/<feature>, the expected path in that case is
+      // "//feature" and not "/feature" (Although "/feature" would be a valid Json Pointer path)
+
+      String lastSegment = jsonPath.substring(lastSegmentPos + 1);
+
+      // First, check if the path ends with an index.
+      Optional<Integer> index = getIndexSegment(lastSegment);
+
+      final String objectURI, featureName;
+
+      if (index.isPresent()) {
+         // Index is present, the path is in the form modeluri#objectfragment/featureName/index
+         String objectAndFeaturePath = jsonPath.substring(0, lastSegmentPos);
+         lastSegmentPos = objectAndFeaturePath.lastIndexOf('/');
+         objectURI = objectAndFeaturePath.substring(0, lastSegmentPos);
+         featureName = objectAndFeaturePath.substring(lastSegmentPos + 1);
+      } else {
+         // Else: no index, we have a path representing Object + Feature
+         objectURI = jsonPath.substring(0, lastSegmentPos);
+         featureName = jsonPath.substring(lastSegmentPos + 1);
+      }
+
+      URI eObjectURI = URI.createURI(objectURI);
+      eObject = getEObject(modelURI, eObjectURI);
+
+      if (eObject == null) {
+         throw new JsonPatchException("Invalid Object path: " + objectURI);
+      }
+
+      EStructuralFeature feature = null;
+      if (featureName != null) {
+         feature = getFeature(eObject, featureName);
+      }
+
+      return new SettingValue(eObject, feature, index);
+   }
+
+   protected EStructuralFeature getFeature(final EObject eObject, final String featureName) throws JsonPatchException {
+      EStructuralFeature feature = eObject.eClass().getEStructuralFeature(featureName);
+      if (feature == null) {
+         throw new JsonPatchException("Invalid Object property: " + featureName);
+      }
+      return feature;
+   }
+
+   protected EditingDomain getEditingDomain(final EObject eObject) {
+      return AdapterFactoryEditingDomain.getEditingDomainFor(eObject);
+   }
+
+   protected Resource getResource(final String modelURI) {
+      return getResource(modelURI, null);
+   }
+
+   protected abstract Resource getResource(String modelURI, URI resourceURI);
+
+   @SuppressWarnings("checkstyle:IllegalExceptionCatch")
+   protected EObject getEObject(final String modelURI, final URI eObjectURI) {
+      Resource resource = getResource(modelURI, eObjectURI.trimFragment());
+      if (resource == null) {
+         return null;
+      }
+      try {
+         return resource.getEObject(eObjectURI.fragment());
+      } catch (Exception ex) {
+         // The object URI is not valid. Ignore the exception and
+         // return null.
+         return null;
+      }
+   }
+
+   protected Command testAction(final ResourceSet resourceSet, final JsonNode patchAction)
+      throws JsonPatchTestException {
+      boolean isValid = true;
+      // TODO ... implement actual test
+      // Should we throw an exception, or return an UnexecutableCommand?
```

Comment:
I think Exception gives us more flexibility, to generate e.g. a more detailed client reply. "UnexecutableCommand" doesn't say anything about why it didn't work; an Exception can hold more valuable information (And we probably want to return a different result based on this exception: e.g. a specific error code/message, rather than "OK but nothing changed" or a generic "Can't execute. Try something else.")

Reply:
I agree that in general an exception may have more information but this exception really does not ;-) Also currently we do not even through an exception. @CamilleLetavernier Could you maybe add a task to track adding this feature? As otherwise this TODO just seems like we were too lazy to do it instead of consciously postponing the implementation.